tfhe/src/high_level_api/config.rs

```diff
@@ -6,14 +6,14 @@ use crate::high_level_api::integers::IntegerConfig;
 use crate::high_level_api::shortints::ShortIntConfig;
 
 /// The config type
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
 pub struct Config {
     #[cfg(feature = "boolean")]
     pub(crate) boolean_config: BooleanConfig,
-    #[cfg(feature = "integer")]
-    pub(crate) integer_config: IntegerConfig,
     #[cfg(feature = "shortint")]
     pub(crate) shortint_config: ShortIntConfig,
+    #[cfg(feature = "integer")]
+    pub(crate) integer_config: Option<IntegerConfig>,
```

Comment:
Because since each type has to be activated via the config, each type may or may not be activated.

So for example, now all integers ( >8bits) are regrouped, into the integer types and the integer type need to be activated via the config. a user could choose not to activate it and instead enable shortint or boolean, so we need Option do handle the optionality of types

Reply:
Do we want to be consistent here maybe ? I don't know how annoying it would be to have the same patter for all 3 here

Reply:
That is the goal but for other PRs,

The plan is:
- To remove shortints to only have integers (and the smallest integer would be 1 block long)
- Refactor the boolean part to simplify it, it was made to function the same way as the shortint and integers part
  even tho it did not needed it.

Reply:
fine by me !